client/firmware.c

```diff
@@ -0,0 +1,262 @@
+/*
+ *	wicked firmware -- utils to firmware config discovery
+ *
+ *	Copyright (C) 2023 SUSE LLC
+ *
+ *	This program is free software; you can redistribute it and/or modify
+ *	it under the terms of the GNU General Public License as published by
+ *	the Free Software Foundation; either version 2 of the License, or
+ *	(at your option) any later version.
+ *
+ *	This program is distributed in the hope that it will be useful,
+ *	but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *	GNU General Public License for more details.
+ *
+ *	You should have received a copy of the GNU General Public License
+ *	along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ *	Authors:
+ *		Marius Tomaschewski
+ *
+ */
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <wicked/util.h>
+#include <wicked/logging.h>
+
+#include "appconfig.h"
+#include "firmware.h"
+#include "process.h"
+#include "buffer.h"
+
+#include <unistd.h>
+#include <getopt.h>
+
+/*
+ * wicked --root-directory <path> option argument in main
+ */
+extern char *		opt_global_rootdir;
+
+static int
+ni_wicked_firmware_extensions(const char *caller, int argc, char **argv)
+{
+	enum {
+		OPTION_HELP	= 'h',
+	};
+	static const struct option	options[] = {
+		{ "help",		no_argument,		NULL,	OPTION_HELP	},
+		{ NULL }
+	};
+	int opt, status = NI_WICKED_RC_USAGE;
+	char *argv0, *command = NULL;
+	const ni_extension_t *ex;
+
+	ni_string_printf(&command, "%s %s", caller, argv[0]);
+	argv0 = argv[0];
+	argv[0] = command;
+	optind = 1;
+	while ((opt = getopt_long(argc, argv, "+h", options, NULL))  != -1) {
+		switch (opt) {
+		case OPTION_HELP:
+			status = NI_WICKED_RC_SUCCESS;
+			/* fall through */
+		default:
+		usage:
```

Comment:
I think we should add the goto jumppoint to the start of the line.
```suggestion
usage:
```

Reply:
This is how we usually do this, see e.g. wicked client main():
```
                switch (c) {
                case OPT_HELP:
                        status = NI_WICKED_ST_OK;
                        /* fall through */
                default:
                usage:
```

Reply:
ok